freeipa/src/main/java/com/sequenceiq/freeipa/converter/cloud/StackToCloudStackConverter.java

```diff
@@ -204,22 +232,24 @@ InstanceTemplate buildInstanceTemplate(Template template, String name, Long priv
                 // existing instances
                 for (InstanceMetaData metaData : instanceGroup.getNotDeletedInstanceMetaDataSet()) {
                     InstanceStatus status = getInstanceStatus(metaData, deleteRequests);
-                    instances.add(buildInstance(metaData, instanceGroup, stackAuthentication, metaData.getPrivateId(), status,
+                    instances.add(buildInstance(stack, metaData, instanceGroup, stack.getStackAuthentication(), metaData.getPrivateId(), status,
                             imageName, environmentCrn));
                 }
                 if (instanceGroup.getNodeCount() == 0) {
-                    skeleton = buildInstance(null, instanceGroup, stackAuthentication, 0L,
+                    skeleton = buildInstance(stack, null, instanceGroup, stack.getStackAuthentication(), 0L,
                             CREATE_REQUESTED, imageName, environmentCrn);
                 }
                 Json attributes = instanceGroup.getAttributes();
-                InstanceAuthentication instanceAuthentication = buildInstanceAuthentication(stackAuthentication);
+                InstanceAuthentication instanceAuthentication = buildInstanceAuthentication(stack.getStackAuthentication());
```

Comment:
Nit: Extract `stack.getStackAuthentication()` as a local variable outside the outer `for` loop.